File: app/src/main/java/invoke/jetters/diceplay/auth_reg/Model/DataBaseConnector.java
package invoke.jetters.diceplay.auth_reg.Model;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.Socket;

public class DataBaseConnector {

    private final BufferedReader reader;
    private final OutputStreamWriter writer;


    public DataBaseConnector() throws IOException {
        Socket clientSocket = new Socket("127.0.0.1", 8080);

        reader = new BufferedReader(
                new InputStreamReader(
                        clientSocket.getInputStream()));

        writer = new OutputStreamWriter(clientSocket.getOutputStream());
    }

    public boolean requestToLogin(User user) throws IOException {

        writer.write("login\n" + user.getLogin() + "\n" + user.getPassword() + "\n");
        writer.flush();

        return reader.readLine().equals("accept");
    }

    public boolean requestToRegistration(User user) throws IOException {
        writer.write("register\n" + user.getLogin() + "\n" + user.getPassword() + "\n");
        writer.flush();

        return reader.readLine().equals("accept");
    }


}

File: app/src/main/java/invoke/jetters/diceplay/auth_reg/Model/DataBaseConnector.kt
package invoke.jetters.diceplay.auth_reg.Model

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.net.Socket
import io.reactivex.Single
import kotlin.jvm.Throws

class DataBaseConnector {
    private val reader: BufferedReader
    private val writer: OutputStreamWriter

    @Throws(IOException::class)
    fun sendRequest(user: User?, requestType: String): Single<Boolean> {
        return Single.create { subscriber ->
            writer.write(""" $requestType ${user?.login} ${user?.password} """.trimIndent())
            writer.flush()
            subscriber.onSuccess(reader.readLine() == "accept");
        }
    }

    init {
        val clientSocket = Socket("127.0.0.1", 8080)
        reader = BufferedReader(
                InputStreamReader(
                        clientSocket.getInputStream()))
        writer = OutputStreamWriter(clientSocket.getOutputStream())
    }
}